fiftyone/core/dataset.py

```diff
@@ -2176,7 +2176,7 @@ def _delete_summary_fields(self, field_names, error_level):
                     error_level,
                 )
             else:
-                if field.read_only:
+                if getattr(field, "read_only", False):
```

Comment:
This change is definitely not necessary. If `field` is a `mongoengine.Field` then `field.save()` will fail too.